unlock-protocol.com/src/components/interface/Membership.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export class Membership extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isMember: 'pending',
+    }
+  }
+
+  componentDidMount() {
+    // Connect to Unlock and get the state!
+    window.addEventListener('unlockProtocol', event => {
+      this.setState(state => ({
+        ...state,
+        isMember: event.detail === 'unlocked' ? 'yes' : 'no',
+      }))
+    })
+  }
+
+  becomeMember() {
+    window.unlockProtocol && window.unlockProtocol.loadCheckoutModal()
+  }
+
+  render() {
+    const { isMember } = this.state
+    if (isMember === 'yes') {
+      return (
+        <MembersBar>
+          Thanks for being a member! Did you know? Your key is an Ethereum non
+          fungible token!{' '}
+          <span role="img" aria-label="click">
+            🔑
+          </span>
+        </MembersBar>
+      )
+    }
+    return (
+      <MembersBar>
+        <Button onClick={this.becomeMember}>
+          Try Unlock, become a member!{' '}
+          <span role="img" aria-label="click">
+            ↗️
+          </span>
+        </Button>
+      </MembersBar>
+    )
+  }
+}
+
+export default Membership
+
+const MembersBar = styled.div`
+  display: grid;
```

Comment:
`display: grid` always needs `grid-template-rows` and `grid-template-columns` definitions

Reply:
Is that true? I could not find any source for this and most of the example that I see out there do not have both of these... (and many actually do not have any...)

Reply:
Without them, the browser does its defaults. Safari adds about 30px padding to rows than chrome. see #3791 for an example

Reply:
huh, ok, what do you suggest I use then?

Reply:
The specific values aren't terribly important (whatever looks good) as long as there's something set to ensure consistency across browsers

Reply:
In this specific case, since it looks like we just need that line of text centered vertically and horizontally, you could even just use a flex container instead of fussing with the rows and columns

Reply:
I agree with @cnasc, flex seems better to use for a 1-column/1-row layout